tests: `TestCrossRefOrcid.test_fetch_references` fails due to brittle assertion logic

### Summary

The test case `test_fetch_references` is failing because of a brittle assertion that expects a specific name prefix (formatted with markdown bolding) to exist within fetched reference strings.

### Error details

```
tests\test_ops\test_apis.py:64 (TestCrossRefOrcid.test_fetch_references[1-work_types1])
self = <tests.test_ops.test_apis.TestCrossRefOrcid object at ...>
co = <pyhelpers.ops.apis.CrossRefOrcid object at ...>
orcid_id = '0000-0002-6502-9934', work_types = ['journal', 'conference paper']
recent_years = 1

    @pytest.mark.parametrize('work_types', [None, ['journal', 'conference paper']])
    @pytest.mark.parametrize('recent_years', [1, 2])
    def test_fetch_references(self, co, orcid_id, work_types, recent_years):
        references = co.fetch_references(
            orcid_id=orcid_id, work_types=work_types, recent_years=recent_years)
        time.sleep(2)
>       assert any(f'**{co.my_name[:5]}' in ref for ref in references)
E       assert False
E        +  where False = any(<generator object TestCrossRefOrcid.test_fetch_references.<locals>.<genexpr> at ...>)

..\pyhelpers\tests\test_ops\test_apis.py:71: AssertionError
```

### Steps to reproduce

```bash
python -m pytest tests/test_ops/test_apis.py::TestCrossRefOrcid::test_fetch_references
```

### Suggested fixes

- Verify if the record contains publications within the `recent_years` filter.

--- a/tests/test_ops/test_apis.py
+++ b/tests/test_ops/test_apis.py
@@ -68,7 +68,8 @@ class TestCrossRefOrcid:
         references = co.fetch_references(
             orcid_id=orcid_id, work_types=work_types, recent_years=recent_years)
         time.sleep(2)
-        assert any(f'**{co.my_name[:5]}' in ref for ref in references)
+        if len(references) > 0:
+            assert any(f'**{co.my_name[:5]}' in ref for ref in references)
 
     @pytest.mark.parametrize('max_entries', [100, 2])
     def test_update_references(self, co, orcid_id, max_entries, tmp_path, monkeypatch, capfd):

